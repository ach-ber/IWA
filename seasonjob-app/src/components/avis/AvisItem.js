import { Pressable, View, StyleSheet, Image, Text } from "react-native";
import Colors from "../../assets/colors/Colors";
import IconAwesome from 'react-native-vector-icons/FontAwesome';
import React from "react";
import RatingStars from "../../utils/RatingStars";

export default function AvisItem({ avis, onpress }) {

    return avis.note ? (
        <Pressable style={styles.avisContainer} onPress={onpress}>
            <View style={styles.imageContainer}>
                <Image src="" style={styles.image} />
            </View>
            <View style={styles.candidatContainer}>
                <Text style={styles.candidatInfo}>{avis.nom} {avis.prenom}</Text>
                <Text>{avis.job}</Text>
            </View>
            <View style={styles.rightContainer}>
                <RatingStars rating={avis.note} color={Colors.yellowStar.color} size={12} style={styles.noteContainer} />
                <Text style={styles.date}>{avis.date}</Text>
            </View>
        </Pressable>
    ) : (
        <Pressable style={styles.avisContainer} onPress={onpress}>
            <View style={styles.imageContainer}>
                <Image src="" style={styles.image} />
            </View>
            <View style={styles.candidatContainer}>
                <Text style={styles.candidatInfo}>{avis.nom} {avis.prenom}</Text>
                <Text>{avis.job}</Text>
            </View>
            <View style={styles.rightContainer}>
                <Text >{avis.date}</Text>
            </View>
        </Pressable>
    );
}

const styles = StyleSheet.create({
    avisContainer: {
        width: "100%",
        height: 100,
        alignItems: 'center',
        justifyContent: 'center',
        borderRadius: 10,
        flexDirection: 'row',
        backgroundColor: 'white',
        marginBottom: 10,
    },
    imageContainer: {
        width: "30%",
        height: "100%",
        alignItems: 'center',
        justifyContent: 'center',
    },
    image: {
        height: 60,
        width: 60,
        borderRadius: 100,
        backgroundColor: 'grey'
    },
    candidatContainer: {
        width: "40%",
        height: "100%",
        alignItems: 'flex-start',
        justifyContent: 'center',
        flexDirection: 'column',
    },
    candidatInfo: {
        fontSize: 20,
        fontWeight: 'bold',
        color: Colors.darkGrey.color,
    },
    rightContainer: {
        width: "30%",
        height: "100%",
        alignItems: 'center',
        justifyContent: 'space-around',
        flexDirection: 'column',
    },
    noteContainer: {
        flexDirection: 'row',
        alignItems: 'flex-end',
        justifyContent: 'space-between',
        width: '70%',
    },
    date: {
        fontSize: 12,
        textAlign: 'left',
        color: Colors.lightGrey.color,
    }
});