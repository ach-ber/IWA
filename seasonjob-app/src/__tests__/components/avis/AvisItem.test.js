import React from "react";
import renderer from "react-test-renderer";
import AvisItem from "../../../components/avis/AvisItem";


describe("AvisItem tests", () => {
    // dumb test to check if jest is working
    test("jest test", () => {
        expect(1).toBe(1);
    })

    test("AvisItem renders correctly", () => {
        const avis = {
            id: 1, nom: "nom", prenom: "prenom", note: 9, job: "Serveur", date: "11/09/2023", titre: "titre avis 1",
            avis: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation",
        }

        const tree = renderer.create(<AvisItem avis={avis} onpress={console.log("pressed")} />).toJSON();
        expect(tree).toMatchSnapshot();
        // on test que le composant ne change pas par rapport à sa version précédente
    })
})