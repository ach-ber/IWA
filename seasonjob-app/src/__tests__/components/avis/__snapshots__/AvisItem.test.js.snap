// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvisItem tests AvisItem renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 10,
      "flexDirection": "row",
      "height": 100,
      "justifyContent": "center",
      "marginBottom": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "width": "30%",
      }
    }
  >
    <Image
      src=""
      style={
        {
          "backgroundColor": "grey",
          "borderRadius": 100,
          "height": 60,
          "width": 60,
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "flex-start",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "center",
        "width": "40%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#111425",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      nom
       
      prenom
    </Text>
    <Text>
      Serveur
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "space-around",
        "width": "30%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "70%",
        }
      }
    >
      <Text />
      <Text />
      <Text />
      <Text />
      <Text />
    </View>
    <Text
      style={
        {
          "color": "#a3a3a3",
          "fontSize": 12,
          "textAlign": "left",
        }
      }
    >
      11/09/2023
    </Text>
  </View>
</View>
`;
